File: LeetCode/Medium/0082-remove-duplicates-from-sorted-list-ii/0082-remove-duplicates-from-sorted-list-ii.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode dummy = new ListNode();
        dummy.next = head;
        ListNode slow = dummy;
        ListNode fast = head;
        
        while (fast != null){
            if (fast.next != null && fast.val == fast.next.val){
                while (fast.next != null && fast.val == fast.next.val) fast.next = fast.next.next;
                fast = fast.next;
                slow.next = fast;
            } else {
                slow = slow.next;
                fast = fast.next;
            }
        }
        
        return dummy.next;
    }
}

File: LeetCode/Medium/0082-remove-duplicates-from-sorted-list-ii/0082-remove-duplicates-from-sorted-list-ii.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteDuplicates(head *ListNode) *ListNode {
    dummy := &ListNode{}
    dummy.Next = head
    slow := dummy
    fast := head
    
    for fast != nil {
        if fast.Next != nil && fast.Val == fast.Next.Val {
            for fast.Next != nil && fast.Val == fast.Next.Val { fast.Next = fast.Next.Next }
            fast = fast.Next
            slow.Next = fast
        } else {
            fast = fast.Next
            slow = slow.Next
        }
    }
    
    return dummy.Next
}